core/include/traccc/edm/measurement.hpp

```diff
@@ -62,22 +62,24 @@ TRACCC_HOST_DEVICE inline bool operator<(const measurement& lhs,
 
     if (lhs.surface_link != rhs.surface_link) {
         return lhs.surface_link < rhs.surface_link;
-    } else if (math::fabs(lhs.local[0] - rhs.local[0]) > float_epsilon) {
-        return (lhs.local[0] < rhs.local[0]);
-    } else {
-        return (lhs.local[1] < rhs.local[1]);
+    } else if (lhs.local[0] != rhs.local[0]) {
+        return lhs.local[0] < rhs.local[0];
+    } else if (lhs.local[1] != rhs.local[1]) {
+        return lhs.local[1] < rhs.local[1];
+    } else if (lhs.variance[0] != rhs.variance[0]) {
+        return lhs.variance[0] < rhs.variance[0];
+    } else if (lhs.variance[1] != rhs.variance[1]) {
+        return lhs.variance[1] < rhs.variance[1];
     }
+    return false;
 }
 
 /// Equality operator for measurements
 TRACCC_HOST_DEVICE
 inline bool operator==(const measurement& lhs, const measurement& rhs) {
 
     return ((lhs.surface_link == rhs.surface_link) &&
-            (math::fabs(lhs.local[0] - rhs.local[0]) < float_epsilon) &&
-            (math::fabs(lhs.local[1] - rhs.local[1]) < float_epsilon) &&
-            (math::fabs(lhs.variance[0] - rhs.variance[0]) < float_epsilon) &&
-            (math::fabs(lhs.variance[1] - rhs.variance[1]) < float_epsilon));
+            (lhs.local == rhs.local) && (lhs.variance == rhs.variance));
```

Comment:
Do we have reasonable definitions for these operators? You're not requiring bit-wise equivalence for these floating point arrays, right? 😕

Reply:
I am meaning to have bit-level equivalence here. This will make CPU and GPU measurements different but I don't think that's a problem. For tolerances with epsilon, `is_same_object` can do exactly the same thing

Reply:
I decided to leave `operator==` as it is. I found that it is going to drop the CKF track_candidate_vector matching rate to 0% as the measurement from CPU and GPU will be different..

The comparison of track candidate element (which is a measurement) just uses the `operator ==` instead of `is_same_object<measurement>`:
https://github.com/acts-project/traccc/blob/8dd31a49ee5bd9fbf2bfc1de24b700a4ae4f99f3/performance/src/performance/details/is_same_object.cpp#L127

I think this line should be using `is_same_object<measurement>` but let me do this in the later PR..